brave/src/main/java/brave/internal/baggage/BaggageHandler.java

```diff
@@ -56,12 +56,18 @@
   S newState(BaggageField field, String value);
 
   /**
-   * Updates a state object to handle a field value change.
+   * Updates a state object to include a value change.
+   *
+   * <p>When the value is {@code null} and the input only includes the given field, either {@code
```

Comment:
personally I think this is a lot simpler if empty coerces to null. A lot less to explain, no chance of accidental mutable empty types, easier tests.. do you see what I mean @anuraaga ?